### DISCUSSION TITLE
How to redirect old URLs in Apache?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#138] [Ask a question] [answered]
If you install Datenstrom Yellow to replace an old website some of your bookmarks might not work anymore. 

Datenstrom Yellow uses [clean URLs](https://en.wikipedia.org/wiki/Clean_URL). For example the RSS feed that was previously available at `http://website/?feed=rss` is now available at `http://website/feed/page:feed.xml`. The blog posting `http://website/post/145454326039.html` is now available at `http://website/blog/hello-world`. You probably want to redirect all URLs with `.html` at the end to the new URL.

What's the best way to redirect old URLs?

**Comment by markseuffert** [comment, marked as answer]
Here's an example `.htaccess` configuration file the Apache web server:

``` apache
<IfModule mod_rewrite.c>
RewriteEngine on

# Redirect old RSS feed
RewriteCond %{QUERY_STRING} feed=rss
RewriteRule ^ /feed/page:feed.xml [R=301,QSD,L]

DirectoryIndex index.html yellow.php
RewriteRule ^(cache|content|system)/ error [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^ yellow.php [L]
</IfModule>
```

Update: Here's a second example for older versions **before Apache 2.4**:

``` apache
<IfModule mod_rewrite.c>
RewriteEngine on

# Redirect old RSS feed
RewriteCond %{QUERY_STRING} feed=rss
RewriteRule ^ /feed/page:feed.xml? [R=301,L]

DirectoryIndex index.html yellow.php
RewriteRule ^(cache|content|system)/ error [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^ yellow.php [L]
</IfModule>
```

**Reply by markseuffert** [reply]
Here's an `.htaccess` example to redirect old blog URLs:

``` apache
<IfModule mod_rewrite.c>
RewriteEngine on

# Redirect old URLs
Redirect 301 /post/152780750859.html /blog/coffe-is-good-for-you
Redirect 301 /post/145454326039.html /blog/hello-world
Redirect 301 /category/computer/ /blog/tag:computer/
Redirect 301 /tag/software/ /blog/tag:software/
Redirect 301 /profile/ /about/

DirectoryIndex index.html yellow.php
RewriteRule ^(cache|content|system)/ error [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^ yellow.php [L]
</IfModule>
```

**Comment by ghost** [comment]
Thanks for the hint, but when I add the two lines. I get this:

> Internal Server Error
> The server encountered an internal error or misconfiguration and was unable to complete your request.

**Comment by markseuffert** [comment]
It works fine on the demo server, which runs Apache/2.4 Linux. See anything in your Apache error log?

**Comment by ghost** [comment]
It seems like that my server has an older version than apache 2.4 :( Therefore the flag QSD is not working.

**Comment by markseuffert** [comment]
Didn't know it, I added a second example. Hope it helps. :smile:

**Comment by ghost** [comment]
That worked! Thanks!
